frontend/src/main/scala/bloop/engine/tasks/Tasks.scala

```diff
@@ -142,10 +142,18 @@ object Tasks {
             List(Config.TestArgument(rawArgs, Some(Config.TestFramework(List(cls)))))
           case frameworks =>
             val frameworkNames = foundFrameworks(frameworks)
-            val ignoredArgs = frameworkSpecificRawArgs.mkString(" ")
-            logger.warn(
+            val (sysProperties, ignoredArgs) = frameworkSpecificRawArgs.partition(s => s.startsWith("-D"))
+
+            if(ignoredArgs.nonEmpty)
+              logger.warn(
               s"Ignored CLI test options '${ignoredArgs}' can only be applied to one framework, found: $frameworkNames")
-            Nil
+
+            val cls = frameworks.map(f => f.getClass.getName)
+
+            if(sysProperties.isEmpty)
```

Comment:
I propose we add the `logger.warn` from above in this part of the `if` expression so that the logic is clearer to follow (we can assume `frameworkSpecificRawArgs.nonEmpty == true` always, which makes `ignoredArgs.nonEmpty == sysProperties.isEmpty`. Also, let's move `cls` to the block of `else` to avoid computing it the arguments are only system properties.

Reply:
if user pass in both system properties and test framework specific args, `ignoredArgs.nonEmpty` and `sysProperties.isEmpty` won't be the same tho

Reply:
Ohhh, I see what you wanted to see here. I apologize for my confusing comment. I'll submit another PR with this change plus an error message if the user passes both system properties and ignored args.